OrbitCore/LinuxUtils.cpp

```diff
@@ -84,41 +88,64 @@ std::string ExecuteCommand(const char* a_Cmd) {
+  std::vector<std::string> result = ListModules(pid);
+
   for (const std::string& line : result) {
-    std::vector<std::string> tokens = Tokenize(line);
-    if (tokens.size() == 6) {
-      const std::string& moduleName = tokens[5];
-
-      auto addresses = Tokenize(tokens[0], "-");
-      if (addresses.size() == 2) {
-        auto iter = modules.find(moduleName);
-        std::shared_ptr<Module> module =
-            (iter == modules.end()) ? std::make_shared<Module>() : iter->second;
-
-        uint64_t start = std::stoull(addresses[0], nullptr, 16);
-        uint64_t end = std::stoull(addresses[1], nullptr, 16);
-
-        if (module->m_AddressStart == 0 || start < module->m_AddressStart)
-          module->m_AddressStart = start;
-        if (end > module->m_AddressEnd) module->m_AddressEnd = end;
-
-        module->m_FullName = moduleName;
-        module->m_Name = Path::GetFileName(module->m_FullName);
-        module->m_Directory = Path::GetDirectory(module->m_FullName);
-        module->m_PdbSize = Path::FileSize(moduleName);
-        modules[moduleName] = module;
-      }
+    std::vector<std::string> tokens =
+        absl::StrSplit(line, " ", absl::SkipEmpty());
+
+    // tokens[4] is the inode column. If inode equals 0, then the memory is not
+    // mapped to a file (might be heap, stack or something else)
+    if (tokens.size() != 6 || tokens[4] == "0") continue;
+
+    const std::string& module_name = tokens[5];
+
+    std::vector<std::string> addresses = absl::StrSplit(tokens[0], "-");
+    if (addresses.size() != 2) continue;
+
+    uint64_t start = 0;
+    std::from_chars(addresses[0].data(),
+                    addresses[0].data() + addresses[0].size(), start, 16);
+    uint64_t end = 0;
+    std::from_chars(addresses[1].data(),
+                    addresses[1].data() + addresses[1].size(), end, 16);
+
+    auto iter = address_map.find(module_name);
+    if (iter == address_map.end()) {
+      address_map[module_name] = {start, end};
+    } else {
+      AddressRange& address_range = iter->second;
+      address_range.start_address =
+          std::min(address_range.start_address, start);
+      address_range.end_address = std::max(address_range.end_address, end);
     }
   }
 
-  for (auto& iter : modules) {
-    auto module = iter.second;
-    module->GetPrettyName();
-    o_ModuleMap[module->m_AddressStart] = module;
+  for (const auto& [module_name, address_range] : address_map) {
+    module_map->insert_or_assign(
+        address_range.start_address,
+        std::make_shared<Module>(module_name, address_range.start_address,
+                                 address_range.end_address));
+
+    std::shared_ptr<Module> module = (*module_map)[address_range.start_address];
+
+    // This filters out entries which are unnaccessible
+    if (module->m_PdbSize == 0) continue;
+
+    std::unique_ptr<ElfFile> elf_file = ElfFile::Create(module->m_FullName);
+    if (!elf_file) continue;
```

Comment:
Is that a regular case that can happen or should this be logged as an ERROR?

Reply:
good point! done